docs/advanced/ttl.md

```diff
@@ -1,9 +1,15 @@
 # Configure DNS record TTL (Time-To-Live)
 
-An optional annotation `external-dns.alpha.kubernetes.io/ttl` is available to customize the TTL value of a DNS record.
-TTL is specified as an integer encoded as string representing seconds.
+> To customize DNS record TTL (Time-To-Live) in a DNS record`, you can use the `external-dns.alpha.kubernetes.io/ttl: <duration>` annotation or flag `--min-ttl=<duration>`. TTL is specified as an integer encoded as string representing seconds. Example; `1s`, `1m2s`, `1h2m11s`
 
-To configure it, simply annotate a service/ingress, e.g.:
+Behaviour:
+
+- If the `external-dns.alpha.kubernetes.io/ttl` annotation is set, it overrides the default TTL(0) value.
+- If the annotation is not set, the default TTL value is used, unless the `--min-ttl` flag is provided.
+- If the annotation is set to `0`, and the `--min-ttl=1s` flag is provided, the value from `--min-ttl` will be used instead.
```

Comment:
On other flags / annotation pair, the annotation (always) override the CLI flag. 

I have doubt about this specific point of the behavior :thinking: . 

@ivankatliarchuk Wdyt ? Do you have a strong motivation behind it that I missed ?

Reply:
Need to review it.

Reply:
So, this is an edge case. Only when value is set to `0`. I have tried to change that, but it will require to change the whole logic behind TTL
1. When TTL is 0, it's classed as not configured https://github.com/kubernetes-sigs/external-dns/blob/07ae419795280b827d0d5c2fc19e6d5b21b5b124/endpoint/endpoint.go#L70. So 0 is the default value, and if annotation is setting it to 0, it will not take any effect.
2. I was thinking to change TTL struct https://github.com/kubernetes-sigs/external-dns/blob/07ae419795280b827d0d5c2fc19e6d5b21b5b124/endpoint/endpoint.go#L239 to not just be `int64`, but the change is quite complex

So I'm proposing to keep this edge case as is, and if we have any feature requests, to revisit this edge case in the future.

Reply:
Basically, at the moment, following two are equal in terms of TTL
```yaml
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: echoserver
  namespace: nm-001
....
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: echoserver
  namespace: nm-001
  annotations:
    external-dns.alpha.kubernetes.io/ttl: "0"
```

One is without TTL annotation and the other one with will TTL annotation set to explicitly to `0`. This case at the moment classed as TTL is not configured.

Reply:
And here https://github.com/kubernetes-sigs/external-dns/pull/320#discussion_r133809237

Basically, when annotation is set to 0, it means use defaults.